src/env_dispatch.cpp

```diff
@@ -299,6 +305,8 @@ static void handle_read_limit_change(const environment_t &vars) {
         } else {
             read_byte_limit = limit;
         }
+    } else {
+        read_byte_limit = DEFAULT_READ_BYTE_LIMIT;
```

Comment:
Nice fix, LGTM.
In future we should probably get rid of most global variables but that doesn't seem relevant now.

Reply:
Oh, definitely, this thing is super janky - and how exactly would that work with multiple parsers?